Mathlib/Combinatorics/SimpleGraph/Basic.lean

```diff
@@ -967,6 +971,22 @@ theorem edgeSet_univ_card : (univ : Finset G.edgeSet).card = G.edgeFinset.card :
   Fintype.card_of_subtype G.edgeFinset fun _ => mem_edgeFinset
 #align simple_graph.edge_set_univ_card SimpleGraph.edgeSet_univ_card
 
+variable [Fintype V] [DecidableEq V]
+
+@[simp]
+theorem edgeFinset_top : (⊤ : SimpleGraph V).edgeFinset = univ.filter fun e => ¬e.IsDiag := by
+  ext f; exact f.inductionOn (by simp)
+
+/-- The complete graph on `n` vertices has `n.choose 2` edges. -/
+theorem card_edgeFinset_top_eq_card_choose_two :
+    (⊤ : SimpleGraph V).edgeFinset.card = (Fintype.card V).choose 2 := by
+  simp only [Set.toFinset_card, edgeSet_top, Set.coe_setOf, ← Sym2.card_subtype_not_diag]
+
+/-- Any graph on `n` vertices has at most `n.choose 2` edges. -/
+theorem card_edgeFinset_le_card_choose_two : G.edgeFinset.card ≤ (Fintype.card V).choose 2 := by
+  apply (card_le_of_subset _).trans_eq card_edgeFinset_top_eq_card_choose_two
+  simp only [Set.subset_toFinset, Set.coe_toFinset, edgeSet_subset_edgeSet, le_top]
```

Comment:
Here's a more conceptual proof. (I golfed this from a few `apply`s that show the steps better, but at least you can see the expected types in the info view to follow along).
```suggestion
  rw [← card_edgeFinset_top_eq_card_choose_two]
  exact card_le_of_subset (edgeFinset_mono le_top)
```

Reply:
Done.